### PR TITLE
Implement results servlet

### PR DESCRIPTION
**Pull request by juanuribe28**
Add a java servlet to serve results.html (previously main.html) given a datastore key as a query parameter. Update javascript and the html to work with this new feature.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/ResultsServlet.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import com.github.mustachejava.DefaultMustacheFactory;
+
+/** Servlet that shows results. */
+@WebServlet("/results")
+public class ResultsServlet extends HttpServlet {
```
Comment (nssafir) [inline src/main/java/com/google/sps/servlets/ResultsServlet.java:16]: [optional] maybe try to name this servlet something more specific to keyword results, unless you are planning on generalizing this logic to show the results for actions and books

File: src/main/java/com/google/sps/servlets/ResultsServlet.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+import java.util.HashMap;
+import com.github.mustachejava.Mustache;
+import com.github.mustachejava.MustacheFactory;
+import com.github.mustachejava.DefaultMustacheFactory;
+
+/** Servlet that shows results. */
```
Comment (nssafir) [inline src/main/java/com/google/sps/servlets/ResultsServlet.java]: [optional] try to be a bit more specific with this comment

File: src/main/java/com/google/sps/servlets/ResultsServlet.java (hunk 1 above)
Comment (juanuribe28) [reply src/main/java/com/google/sps/servlets/ResultsServlet.java:16]: Well, it is serving the template that is used to render keywords, actions and books.

Comment (nssafir) [reply src/main/java/com/google/sps/servlets/ResultsServlet.java:16]: Nvm then

**Review by nssafir — APPROVED** [review]

**Review by seangilpin — APPROVED** [review]

**Pull request merged**